fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java

```diff
@@ -124,16 +134,45 @@ public void createTable(CreateTableStmt stmt) throws UserException {
             throw new UserException("Failed to get database: '" + dbName + "' in catalog: " + catalog.getName());
         }
         try {
-            Map<String, String> props = stmt.getExtProperties();
-            String fileFormat = props.getOrDefault("file_format", Config.hive_default_file_format);
-            HiveTableMetadata catalogTable = HiveTableMetadata.of(dbName,
-                    tblName,
-                    stmt.getColumns(),
-                    parsePartitionKeys(props),
-                    props,
-                    fileFormat);
-
-            client.createTable(catalogTable, stmt.isSetIfNotExists());
+            Map<String, String> props = stmt.getProperties();
+            String fileFormat = props.getOrDefault(FILE_FORMAT_KEY, Config.hive_default_file_format);
+            Map<String, String> ddlProps = new HashMap<>();
+            for (Map.Entry<String, String> entry : props.entrySet()) {
+                String key = entry.getKey().toLowerCase();
+                if (DORIS_HIVE_KEYS.contains(entry.getKey().toLowerCase())) {
+                    ddlProps.put("doris." + key, entry.getValue());
+                } else {
+                    ddlProps.put(key, entry.getValue());
+                }
+            }
+            List<String> partitionColNames = new ArrayList<>();
+            if (stmt.getPartitionDesc() != null) {
+                partitionColNames.addAll(stmt.getPartitionDesc().getPartitionColNames());
+            }
+            HiveTableMetadata hiveTableMeta;
+            DistributionDesc bucketInfo = stmt.getDistributionDesc();
+            if (bucketInfo != null && Config.enable_external_hive_bucket) {
```

Comment:
This if else is wrong, if Config.enable_external_hive_bucket is false and bucketInfo != null, should throw exception